src/api/app/controllers/webui2/repositories_controller.rb

```diff
@@ -7,7 +7,7 @@ def webui2_index
 
     @user_can_set_flags = policy(@project).update?
 
-    @architectures = @project.architectures.reorder('name')
+    @architectures = Architecture.where(id: @project.repository_architectures.select(:architecture_id)).reorder('name').distinct
```

Comment:
why is the distinct needed? :confused:

Reply:
it's most likely not, I just wasn't sure